src/compiler/checker.ts

```diff
@@ -10920,7 +10920,7 @@ namespace ts {
             target: Signature,
             ignoreReturnTypes: boolean): boolean {
             return compareSignaturesRelated(source, target, CallbackCheck.None, ignoreReturnTypes, /*reportErrors*/ false,
-                /*errorReporter*/ undefined, compareTypesAssignable) !== Ternary.False;
+                /*errorReporter*/ undefined, compareTypesAssignable, compareTypesIdentical) !== Ternary.False;

```

Comment:
Rather than `compareTypesIdentical`, use `compareTypes(A, B) && compareTypes(B,A)`. This should handle `any` much better and actually maps better to what "invariance" means, unlike `compareTypesIdentical`.

Reply:
(I suggest this because I believe it'll eliminate most of the RWC breaks)

Reply:
Done.  (It's weird to me that apparently our concept of "invariance" allows one method type to be replaced by another that differs in a bivariant parameter, but that's an unrelated issue.)  It looks like this will eliminate the mobx break and maybe the rxjs break, but not the VS Code break or the @types/vinyl break.